superdesk/sequences.py

```diff
@@ -1,37 +1,91 @@
+from .resource import Resource
+from .services import BaseService
+from flask import current_app as app
 
 
-def get_next_sequence_number(
-    resource_name, item_id,
-    max_seq_number, min_seq_number=0,
-    field_name='sequence_number'
-):
-    """
-    Generates Sequence Number
-    :param resource_name: target resource
-    :param item_id: target item id
-    :param max_seq_number: maximal possible value
-    :param min_seq_number: default 0, minimal possible value
-    :param field_name: default 'sequence_number', name of field to store sequence_number in the target item
-    :returns: sequence_number
-    """
-
-    target_resource = get_resource_service(resource_name)
-    sequence_number = target_resource.find_one(
-        req=None,
-        _id=item_id
-    ).get(field_name) or min_seq_number
-    if (min_seq_number <= sequence_number + 1 <= max_seq_number):
-        target_resource.find_and_modify(
-            query={'_id': item_id},
-            update={'$inc': {field_name: 1}},
-            upsert=False
-        )
-    else:
-        target_resource.find_and_modify(
-            query={'_id': item_id},
-            update={field_name: min_seq_number},
-            upsert=False
-        )
+def init_app(app):
+    endpoint_name = 'sequences'
+    service = SequencesService(endpoint_name, backend=superdesk.get_backend())
+    SequencesResource(endpoint_name, app=app, service=service)
+
+
+class SequencesResource(Resource):
+    schema = {
+        'name': {
+            'type': 'string',
+            'required': True,
+            'nullable': False,
+            'empty': False,
+            'iunique': True
+        },
+        'sequence_number': {
+            'type': 'number',
+            'default': 0,
+        },
+    }
+    etag_ignore_fields = ['sequence_number', 'name']
+    internal_resource = True
+
+
+class SequencesService(BaseService):
 
-    return sequence_number
+    def get_next_sequence_number_for_item(
+        self,
+        resource_name,
+        query_value,
+        max_seq_number=None,
+        min_seq_number=0,
+        query_field='_id',
+        sequence_field='sequence_number'
+    ):
+        """
+        Generates Sequence Number
+        :param resource_name: target resource
+        :param query_value: value of field to query target item (ie target item id by default)
+        :param query_field: default '_id', name of field to query target item
+        :param max_seq_number: default $MAX_VALUE_OF_PUBLISH_SEQUENCE, maximal possible value
+        :param min_seq_number: default 0, minimal possible value
+        :param sequence_field: default 'sequence_number', name of field to store sequence number in the target item
+        :returns: sequence_number
+        """
+        if not max_seq_number:
+            max_seq_number = app.config['MAX_VALUE_OF_PUBLISH_SEQUENCE']
+
+        target_resource = get_resource_service(resource_name)
+        target_item = target_resource.find_one(**{
```

Comment:
You don't need to create a dict here, the call should look like:
`find_one(req=None, query_field=query_value)`